doc/man1/genrsa.pod

```diff
@@ -96,6 +97,14 @@ RSA key, which is defined in RFC 8017.
 The size of the private key to generate in bits. This must be the last option
 specified. The default is 2048.
 
+=item B<-insecure>
+
+By default OpenSSL will force users to generate secure multi-prime keys only.
+That is to say, the number of primes is fixed due to a security policy, see
```

Comment:
Is it fixed or just limited above?

Overall, I'd like to see a bit more text explaining why multi-prime keys are weaker.  We're expecting them to make an informed choice without providing any hint as to the trade offs involved.

All that said, the code looks fine.